tests/conftest.py

```diff
@@ -36,9 +36,14 @@ def actor_restart_limit(limit: int) -> Iterator[None]:
         f"<actor_restart_limit> Changing the restart limit from {original_limit} to {limit}"
     )
     _actor.Actor._restart_limit = limit
-    yield
-    print(f"<actor_restart_limit> Resetting restart limit to {original_limit}")
-    _actor.Actor._restart_limit = original_limit
+    try:
+        yield
+    except Exception as e:  # pylint: disable=broad-except
+        print(f"<actor_restart_limit> Unhandled exception caught: {e}")
+        raise
```

Comment:
Not sure if this is providing anything useful, if the exception is re-raised, probably `pytest` should catch it and show all the context of the failure anyway.